tasks/windows/install-package-chocolatey.yml

```
@@ -1,8 +1,38 @@
 ---
+- name: Get info for a single service
+  ansible.windows.win_service_info:
+    name: windows_exporter
+  register: service_info
+  ignore_errors: true
+
+- name: Set expected_textfile_parameter
+  ansible.builtin.set_fact:
+    expected_textfile_parameter: >-
+      --collector.textfile.directory {{ node_exporter_textfile_dir }}
+  when: >
+    node_exporter_textfile_dir is defined
+    and service_info.exists
+
+- name: Set windows_exporter_force_uninstall
+  ansible.builtin.set_fact:
+    windows_exporter_force_uninstall: true
+  when: >
+    node_exporter_textfile_dir is defined
+    and service_info.exists
+    and not expected_textfile_parameter in service_info.services[0].path
+
+- name: Uninstall windows-exporter package
+  chocolatey.chocolatey.win_chocolatey:
+    name: prometheus-windows-exporter.install
+    state: absent
+  when: windows_exporter_force_uninstall
```

If this variable isn't defined, Ansible will freak out. You have three options to solve it:

1. Define this to as `false` in the role's `defaults/main.yml`
2. Add a `set_fact` task to initialize to the default value above the "Set windows_exporter_force_uninstall fact" task
3. Apply this suggestion:

```suggestion
  when:
    - windows_exporter_force_uninstall is defined
    - windows_exporter_force_uninstall
```
```

---

Thanks for pointing this out and giving me some options to chose from 👍 
After some consideration I chose the third option, because it's clear and simple.